### PR TITLE
Add unit tests for b-productswithinputdevices block

### PR DESCRIPTION
**Pull request by Matei-Iordache**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--www-landing-pages--bitdefender.hlx.page/drafts/test-page/
- After: https://DEX-16967--www-landing-pages--bitdefender.hlx.page/drafts/test-page/

### CODE REVIEW

File: _src-lp/blocks/b-productswithinputdevices/b-productswithinputdevices.js [hunk 1]
```diff
@@ -115,7 +115,7 @@ export default function decorate(block) {
 
     block.querySelectorAll('fieldset button').forEach((item) => {
       item.addEventListener('click', () => {
-        const action = item.innerText;
+        const action = item.innerHTML;
```
Comment (vradulescu-bd) [inline _src-lp/blocks/b-productswithinputdevices/b-productswithinputdevices.js]: yes, this is caused by jsdom, better if you change it but keep the initial consistent way: use item.textContent

File: _src-lp/blocks/b-productswithinputdevices/__TESTS__/b-productswithinputdevices.spec.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      
+      // input field is 11 from the previous test, so we click 2 times to get to 9
+      minusButton.click();
+      minusButton.click();
+      expect(inputField.value).toBe('9');
+    });
+
+
+  });
+
```
Comment (vradulescu-bd) [inline _src-lp/blocks/b-productswithinputdevices/__TESTS__/b-productswithinputdevices.spec.js]: remove spacing

File: _src-lp/blocks/b-productswithinputdevices/__TESTS__/b-productswithinputdevices.spec.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      const minusButton = b_productswithinputdevicesContainer.querySelector('.b-productswithinputdevices fieldset button');
+      const inputField = b_productswithinputdevicesContainer.querySelector('.b-productswithinputdevices fieldset input');
+      
+      // input field is 11 from the previous test, so we click 2 times to get to 9
+      minusButton.click();
+      minusButton.click();
+      expect(inputField.value).toBe('9');
+    });
+
+
```
Comment (vradulescu-bd) [inline _src-lp/blocks/b-productswithinputdevices/__TESTS__/b-productswithinputdevices.spec.js]: remove spacing

File: _src-lp/blocks/b-productswithinputdevices/__TESTS__/b-productswithinputdevices.spec.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
+// eslint-disable-next-line import/extensions
+import mockData from './mock-data.json';
+// mock everything that is not related to the actual main scope
+jest.mock('../../../scripts/lib-franklin.js', () => ({
+  ...jest.requireActual('../../../scripts/lib-franklin.js'),
+  loadHeader: jest.fn(),
+  loadFooter: jest.fn(),
+}));
+
+// import { updateTableElement } from '../b-productswithinputdevices.js';
```
Comment (vradulescu-bd) [inline _src-lp/blocks/b-productswithinputdevices/__TESTS__/b-productswithinputdevices.spec.js]: remove unnecessary code

File: _src-lp/blocks/b-productswithinputdevices/__TESTS__/b-productswithinputdevices.spec.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    test('Check when clicking the - button that the value updates correctly', () => {
+      const minusButton = b_productswithinputdevicesContainer.querySelector('.b-productswithinputdevices fieldset button');
+      const inputField = b_productswithinputdevicesContainer.querySelector('.b-productswithinputdevices fieldset input');
+      
+      // input field is 11 from the previous test, so we click 2 times to get to 9
+      minusButton.click();
+      minusButton.click();
+      expect(inputField.value).toBe('9');
+    });
+
```
Comment (vradulescu-bd) [inline _src-lp/blocks/b-productswithinputdevices/__TESTS__/b-productswithinputdevices.spec.js]: pls add one more use case in order to be 100% bullet proof: the component should call the updateProductsList() method ( or assert that window.productsListCount has been changed )

Comment (Matei-Iordache) [reply _src-lp/blocks/b-productswithinputdevices/__TESTS__/b-productswithinputdevices.spec.js]: Added a new test case which covers this case.

File: _src-lp/blocks/b-productswithinputdevices/__TESTS__/b-productswithinputdevices.spec.js (hunk 4 above)
Comment (Matei-Iordache) [reply _src-lp/blocks/b-productswithinputdevices/__TESTS__/b-productswithinputdevices.spec.js]: Done

File: _src-lp/blocks/b-productswithinputdevices/__TESTS__/b-productswithinputdevices.spec.js (hunk 3 above)
Comment (Matei-Iordache) [reply _src-lp/blocks/b-productswithinputdevices/__TESTS__/b-productswithinputdevices.spec.js]: Done

File: _src-lp/blocks/b-productswithinputdevices/__TESTS__/b-productswithinputdevices.spec.js (hunk 2 above)
Comment (Matei-Iordache) [reply _src-lp/blocks/b-productswithinputdevices/__TESTS__/b-productswithinputdevices.spec.js]: Done

File: _src-lp/blocks/b-productswithinputdevices/b-productswithinputdevices.js (hunk 1 above)
Comment (Matei-Iordache) [reply _src-lp/blocks/b-productswithinputdevices/b-productswithinputdevices.js]: Done

**Review by vradulescu-bd — APPROVED** [review]

**Review by enake — APPROVED** [review]

**Pull request merged**